### PR TITLE
Add github workflow for sonarcloud analysis

### PR DESCRIPTION
**Pull request by tomasz-kaminski-sonarsource**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          unzip -o $HOME/.sonar/build-wrapper-linux-x86.zip -d $HOME/.sonar/
+          echo "$HOME/.sonar/build-wrapper-macosx-x86" >> $GITHUB_PATH
+      - name: Run build-wrapper
+        run: |
+          build-wrapper-macosx-x86 --out-dir build_wrapper_output_directory xcodebuild
+      - name: Run sonar-scanner
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+          SONAR_ARGS: "--define sonar.projectVersion=1.0 --define sonar.sources=src --define sonar.cfamily.build-wrapper-output=build_wrapper_output_directory --define sonar.sourceEncoding=UTF-8"
```
Comment (arseniy-sonar) [inline .github/workflows/build.yml]:
```suggestion
          SONAR_ARGS: "--define sonar.projectVersion=1.0 --define sonar.sources=src --define sonar.sourceEncoding=UTF-8"  
```
move this to sonar-project.properties.
See my comment on the windows PR

File: README.adoc [hunk 2]
```diff
@@ -1,3 +1,32 @@
...
+.. Wrap your compilation with the Build Wrapper
+.. Run `sonar-scanner` as the final step
+. Make sure that you have your token stored as a secret in your repository (`SONAR_TOKEN`). You can request new tokens using https://sonarcloud.io/account/security/[Account/Security page].
+
+You can take a look at the link:sonar-project.properties[sonar-project.properties] and link:.github/workflows/build.yml[build.yml] to see it in practice.
+
+= Documentation
+
+- https://docs.sonarqube.org/latest/analysis/languages/cfamily/[Documentation of the C/C++/Objective-C plugin and its Build Wrapper]
+-  https://sonarcloud.io/documentation/getting-started/github/[Documentation of using SonarCloud with GitHub]
```
Comment (arseniy-sonar) [inline README.adoc]:
```suggestion
- https://sonarcloud.io/documentation/getting-started/github/[Documentation of using SonarCloud with GitHub]
```

File: .github/workflows/build.yml (hunk 1 above)
Comment (arseniy-sonar) [reply .github/workflows/build.yml]: Alternatively, remove the duplicating option from the file

Comment (tomasz-kaminski-sonarsource) [reply .github/workflows/build.yml]: I do not understand the second suggestion. 
I disagree, with moving it in the file, as int is generated in two places in this file.

Comment (tomasz-kaminski-sonarsource) [reply .github/workflows/build.yml]: Updated.

**Review by arseniy-sonar — APPROVED** [review]

**Pull request merged**